lib/sla.js

```diff
@@ -1,6 +1,93 @@
 'use strict';
+var Q = require('q');
 
-module.exports = function(callback){
-    console.log("\"snarflegluff\" should be printed below:");
-    var str = "snarflegluff";
-    return str;}
+/**
+ * Looks for an existing mixing bridge and creates one if none exist.
+ * @param {Object} client - Object that contains information from the ARI
+ *   connection.
+ * @return {Q} Q promise object.
+ */
+function designateMixingBridge(client) {
+  var defer = Q.defer();
+  var list = Q.denodeify(client.bridges.list.bind(client.bridges));
```

Comment:
You actually want to bind to client here, not client.bridges. Any time you denodeify a function called on client.someResource you pass the client to bind. Any time you denodeify a function called on a resource instance like bridge.get you pass the bridge to bind.